baybe/campaign.py

```diff
@@ -261,6 +271,80 @@ def recommend(
 
         return rec
 
+    def posterior(self, candidates: pd.DataFrame) -> Posterior:
+        """Get the posterior predictive distribution for the given candidates.
+
+        The predictive distribution is based on the surrogate model of the last used
+        recommender.
+
+        Args:
+            candidates: The candidate points in experimental recommendations.
+                For details, see :meth:`baybe.surrogates.base.Surrogate.posterior`.
+
+        Raises:
+            IncompatibilityError: If the underlying surrogate model exposes no
+                method for computing the posterior distribution.
+
+        Returns:
+            Posterior: The corresponding posterior object.
+            For details, see :meth:`baybe.surrogates.base.Surrogate.posterior`.
+        """
+        surrogate = self.get_surrogate()
+        if not hasattr(surrogate, method_name := "posterior"):
+            raise IncompatibilityError(
+                f"The used surrogate type '{surrogate.__class__.__name__}' does not "
+                f"provide a '{method_name}' method."
+            )
+
+        import torch
+
+        with torch.no_grad():
+            return surrogate.posterior(candidates)
+
+    def get_surrogate(self) -> SurrogateProtocol:
+        """Get the current surrogate model.
+
+        Raises:
+            RuntimeError: If the current recommender does not provide a surrogate model.
+
+        Returns:
+            Surrogate: The surrogate of the current recommender.
+        """
+        # TODO: remove temporary restriction when target transformations can be handled
+        match self.objective:
+            case SingleTargetObjective(
+                _target=NumericalTarget(bounds=b)
+            ) if not b.is_bounded:
+                pass
+            case _:
+                raise NotImplementedError(
+                    "Surrogate model access is currently only supported for a single "
+                    "untransformed target."
+                )
```

Comment:
@Scienfitz @AVHopp: Damn it, I just realized I did not think of all cases, in particular:
1. The fact that even regular MIN/MAX targets can have transforms
2. Nonlinear target transformations like bell
3. DesirabilityObjectives
4. The upcoming BinaryTarget

So right now I'm just gonna restrict the function to the most common case of regular single untransformed targets, which already helps a lot. I'll then need to rethink what is the best way for the other cases. My current feeling:
1. Can be solved easily
2. Not sure at the moment since not invertible
3. No way to disentangle into individual targets. But perhaps just exposing the desirability surrogate is just fine
4. Probably already supported but won't commit to this until the branch is actually merged.

Let me know if you have a better (fast-to-implement) idea/workaround.

Reply:
@Scienfitz, good news, I've though about it a little more and think have a solid solution for all current (and also future cases):

 The current problem that causes the issues is that we do manual transformations on the target values *first* and only *then* build the model on the transformed values. Logically, that was exactly the same issue, btw, that I solved in #340 just for the "MIN" transform. And the solution is also the same: Instead of building the model on the transformed targets and then running into problems of not being able to invert the transformation (like with "BELL"), we should simply by default build the model on the actual targets and then transform the obtained posterior. That is, for example, the desirability averaging would happen at a later stage using botorch's transforms. Plus it comes with the benefit that we can offer surrogates for both the raw targets as well as derived values. So in future we can make the `DesirabilityObjective` return `N` surrogates for the `N` targets involved plus an `N+1` surrogate modeling the desirability posterior values.

That said, this can be nicely wrapped into a separate PR as preparation before tackling multi-targets when I'm back.

So I think that also answer your question in 3). And for 4): I think we can enable it, but I'll have to do it (if time still permits) in the Bandit PR, since I don't even have access to the necessary objects in the current PR.

Let me know if this all makes sense

Reply:
it seems all reasonable to me
but I would like to keep this discussion on whats needed for the PR, and whats outside (and perhaps needs to be refined/duscissed further) belongs into an issue

Reply:
#373